src/main/java/net/imglib2/algorithm/grid/Patch.java

```diff
@@ -0,0 +1,278 @@
+	 * @param span
+	 *            span of each patch: size = 2 * span + 1
+	 * @param whichDims
+	 *            array contains zeros and ones and defines which dimensions the
+	 *            patches have
+	 */
+	public Patch(RandomAccessibleInterval<T> srcImage, long[] gap, long[] span, int[] whichDims) {
+		this.srcImage = srcImage;
+		this.gap = gap;
+		this.span = span;
+		if (whichDims[0] == -1) {
+			whichDims = new int[srcImage.numDimensions()];
+			Arrays.fill(whichDims, 1);
+		}
+		this.whichDims = whichDims;
+
+		update(new long[srcImage.numDimensions()], gap);
+	}
+
+	public Patch(RandomAccessibleInterval<T> srcImage, long[] gap, long[] span) {
+		new Patch<>(srcImage, gap, span, new int[] { -1 });
+	}
+
+	public Patch(RandomAccessibleInterval<T> srcImage, long gap, long span, int[] whichDims) {
+		long[] gapA = new long[srcImage.numDimensions()];
+		long[] spanA = new long[srcImage.numDimensions()];
+		Arrays.fill(gapA, gap);
+		Arrays.fill(spanA, span);
+		new Patch<>(srcImage, gapA, spanA, whichDims);
+	}
+
+	public Patch(RandomAccessibleInterval<T> srcImage, long gap, long span) {
+		new Patch<>(srcImage, gap, span, new int[] { -1 });
+	}
+
+	public long[] getPosition() {
+		return pos;
+	}
+
+	public RandomAccessibleInterval<T> getPatch() {
+		return patch;
+	}
+
+	@Override
+	public RandomAccess<T> randomAccess() {
+		return patch.randomAccess();
+	}
+
+	@Override
+	public RandomAccess<T> randomAccess(Interval interval) {
+		return patch.randomAccess(interval);
+	}
+
+	/**
+	 * 
+	 * @param gridPos
+	 *            position of the grid where the patch should be placed
+	 * @param origin
+	 *            origin of the gird (left top)
+	 */
+	public void update(final long[] gridPos, long[] origin) {
+		// compute the actual position of the patch in the coordinates of the
+		// source image
+		long[] patchPos = new long[gridPos.length];
+		for (int i = 0; i < gridPos.length; i++) {
+			patchPos[i] = origin[i] + gridPos[i] * gap[i];
+		}
+		// compute min and max of the interval of the patch
+		long[] min = new long[patchPos.length];
+		long[] max = new long[patchPos.length];
+		for (int i = 0; i < patchPos.length; i++) {
+			if (whichDims[i] == 1) {
+				min[i] = patchPos[i] - span[i];
+				max[i] = patchPos[i] + span[i];
+			} else {
+				min[i] = patchPos[i];
+				max[i] = patchPos[i];
+			}
+		}
+		this.pos = patchPos;
+
+		// check if the center of the patch is inside the interval of the source
+		// image
+		RandomAccessibleInterval<T> raiPos = Views.interval(srcImage, patchPos, patchPos);
+		if (!Intervals.contains(srcImage, raiPos)) {
+			throw new IndexOutOfBoundsException("IndexOutOfBoundsException");
+		}
+		RandomAccessibleInterval<T> raiPatch = Views.interval(srcImage, min, max);
+		// extend the source image if the patch and borders overlap
+		if (!Intervals.contains(srcImage, raiPatch)) {
+			raiPatch = Views.interval(Views.extendMirrorSingle(srcImage), min, max);
+		}
+
+		this.patch = Views.zeroMin(raiPatch);
```

Comment:
A new `View` is generated whenever `Patch` is returned by `GridRandomAccess.get()`.  This makes the `Patch` class not only obsolete but redundant.  Try to implement a solution that returns `Views.offsetInterval(long[], long[]);` directly in `GridRandomAccess.get()`.